src/device-api/v2.ts

```diff
@@ -470,4 +473,43 @@ export function createV2Api(router: Router, applications: ApplicationManager) {
 			});
 		}
 	});
+
+	router.post('/v2/journal-logs', (req, res) => {
+		let statusSent = false;
+		try {
+			const follow = checkTruthy(req.body.follow) || false;
+			const count = checkInt(req.body.count, { positive: true }) || undefined;
+			const unit = req.body.unit;
+
+			const journald = getJournald({ follow, count, unit });
+
+			journald.on('logLine', msg => {
+				if (!statusSent) {
+					res.status(200);
+					statusSent = true;
+				}
+				res.write(`${JSON.stringify(msg)}\n`);
+			});
+			journald.on('stderrLine', line => {
+				if (!statusSent) {
+					res.status(500);
+				}
+				res.json({
+					status: 'failed',
+					message: `Journald stderr: ${line}`,
+				});
+			});
+			res.on('close', () => {
+				journald.end();
```

Comment:
So this closes journald when response stream is closed.

But when `follow` is set to `true`, we should probably also react on `req` events.
https://nodejs.org/api/http.html#http_event_abort

Still relatively new to nodejs, so may be mistaking. :)

Reply:
`res` is closed when the client closes the connection too, what other events are you interested in catching?

Reply:
I'm good in such case :)